extensions/panache/hibernate-orm-rest-data-panache/deployment/src/test/java/io/quarkus/hibernate/orm/rest/data/panache/deployment/CustomPathItemsResource.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.hibernate.orm.rest.data.panache.deployment;
+
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import io.quarkus.hibernate.orm.rest.data.panache.PanacheEntityCrudResource;
+import io.quarkus.rest.data.panache.OperationProperties;
+import io.quarkus.rest.data.panache.ResourceProperties;
+
+@ResourceProperties(path = "custom-items", hal = true)
+public interface CustomPathItemsResource extends PanacheEntityCrudResource<Item, Long> {
+
+    @OperationProperties(path = "api")
```

Comment:
Given that you've called the class annotation `ResourceProperties` after JAX-RS's (horrible) choice, is there a colloquial term for methods too or did you pick `Operation` at random?

FTR JAX-RS calls methods "resource methods":

> Resource methods are methods of a resource class annotated with a request method designator. They are
used to handle requests and MUST conform to certain restrictions described in this section.
A request method designator is a runtime annotation that is annotated with the @HttpMethod annotation.
JAX-RS defines a set of request method designators for the common HTTP methods: @GET, @POST, @PUT,
@DELETE, @PATCH, @HEAD and @OPTIONS. Users may define their own custom request method designators
including alternate designators for the common HTTP methods

I just want to make sure people are not going to object at us calling this an `Operation`.

Reply:
I called them operations, because that term seems to be used in W3C https://www.w3.org/2001/sw/wiki/REST#Operations_on_Elements. I'm not a purist of any terminology. So whichever is more acceptable I'm fine with it.

Reply:
If method is closer to JAX-RS which is what we're using I'm fine with it

Reply:
I'm only mentioning it because we started with “controller” and we got forced to adhere to JAX-RS's (mistaken) “resource”. So if we need to map all the way to JAX-RS probably it should be `@MethodProperties`?

BTW, that link you posted is non-normative. Reference for REST is https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm (and doesn't mention operations at all. I couldn't find at a glance how it refers to the operations you can do on the resources) but seems to call them “operations”.

Reply:
Yeah I'll just rename to `@MethodProperties` to be inline with JAX-RS. I'm renaming the interface anyway, so will do it at the same time.

Reply:
Done